### PR TITLE
Slightly better gc behavior in some circumstances

### PR DESCRIPTION
**Pull request by SmilyOrg**
... and an out of memory warning

### CODE REVIEW

File: loom/script/native/core/system/lmGC.cpp [hunk 1]
```diff
@@ -159,7 +159,7 @@ class GC
         // it can usually have more leeway as the rate will adjust
         // under normal conditions and only sudden changes benefit from
         // a faster response.
-        bool cycleKBWarn = lastValidBPR > 0 && cycleKBDelta > cycleKB * (hibernating ? targetGarbage : cycleMemoryGrowthWarningRatio);
+        bool cycleKBWarn = cycleKB > 0 && cycleKBDelta > cycleKB * (hibernating ? targetGarbage : cycleMemoryGrowthWarningRatio);
```
Comment (bengarney) [inline loom/script/native/core/system/lmGC.cpp]: This is a horrific expression. Can we break it up into ifs and elses?

**Comment by bengarney** [comment]
One nit.

File: loom/script/native/core/system/lmGC.cpp (hunk 1 above)
Comment (SmilyOrg) [inline loom/script/native/core/system/lmGC.cpp]: It's just three conditions and a multiplication, but fine, I can make it clearer :P

**Comment by SmilyOrg** [comment]
Changed a bit and rebased.

**Comment by bengarney** [comment]
That's still pretty chunky. The in-ternary expression is also needlessly complex. For instance wouldn't

```cycleKB*(hibernating ? targetGarbage : cycleMemoryGrowthWarningRatio);```

be clearer? I want to make sure this logic is pristine and easy to follow because it can have such a colossal impact on performance/app behavior.

**Comment by SmilyOrg** [comment]
Well, that's what I had before! :)

**Comment by SmilyOrg** [comment]
It also has two paragraphs explaining it above the line.

**Pull request merged**